src/ucp/core/ucp_ep.c

```diff
@@ -413,13 +420,139 @@ static ucs_status_t ucp_ep_create_to_sock_addr(ucp_worker_h worker,
+    ucp_ep_add_connected_lane(ucp_ep, uct_ep);
+    ucp_ep_flush_state_reset(ucp_ep);
+}
+
+static void
+ucp_ep_sockaddr_server_connect_cb(uct_ep_h ep, void *arg, ucs_status_t status)
+{
+    ucp_ep_h         ucp_ep = (ucp_ep_h)arg;
+    ucp_lane_index_t wireup_idx;
+
+    if (status != UCS_OK) {
+        wireup_idx = ucp_ep_config(ucp_ep)->key.wireup_lane;
+        ucp_worker_set_ep_failed(ucp_ep->worker, ucp_ep,
+                                 ucp_ep->uct_eps[wireup_idx], wireup_idx,
+                                 status);
+        return;
+    }
+
+    ucp_ep_sockaddr_cm_wireup_complete(ucp_ep, ep);
+}
+
+static ucs_status_t
+ucp_ep_sockaddr_server_create_connected(ucp_worker_h worker,
+                                        const ucp_ep_params_t *params,
+                                        const ucp_unpacked_address_t *remote_address,
+                                        ucp_ep_h *ep_p)
+{
+    ucp_ep_h ep;
+    ucp_wireup_ep_t *wireup_ep;
+    uct_ep_params_t uct_ep_params;
+    char peer_name[UCS_SOCKADDR_STRING_LEN];
+    ucs_status_t status;
+    ucp_md_index_t i;
+
+    ucs_assert(params->field_mask & (UCP_EP_PARAM_FIELD_SOCK_ADDR |
+                                     UCP_EP_PARAM_FIELD_CONN_REQUEST));
+    if (params->field_mask & UCP_EP_PARAM_FIELD_SOCK_ADDR) {
+        ucs_sockaddr_str(params->sockaddr.addr, peer_name,
+                         UCS_SOCKADDR_STRING_LEN);
+    } else {
+        snprintf(peer_name, UCS_SOCKADDR_STRING_LEN, "%p",
+                 (void *)params->conn_request->client_data.ep_ptr);
+    }
+
+    status = ucp_ep_new(worker, peer_name, "conn_request on uct_listener", &ep);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    status = ucp_ep_init_create_wireup(ep, params, &wireup_ep);
+    if (status != UCS_OK) {
+        goto err_del_ep;
+    }
+
+    status = ucp_ep_adjust_params(ep, params);
+    if (status != UCS_OK) {
+        goto err_del_ep;
+    }
+
+    uct_ep_params.field_mask = UCT_EP_PARAM_FIELD_CM                    |
+                               UCT_EP_PARAM_FIELD_CONN_REQUEST          |
+                               UCT_EP_PARAM_FIELD_USER_DATA             |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS     |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB      |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_CONNECT_CB   |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB;
+
+    uct_ep_params.user_data                  = ep;
+    uct_ep_params.conn_request               = params->conn_request->uct_req;
+    uct_ep_params.sockaddr_cb_flags          = UCT_CB_FLAG_ASYNC;
+    uct_ep_params.sockaddr_pack_cb           = ucp_wireup_sockaddr_cm_priv_pack_cb;
+    uct_ep_params.sockaddr_connect_cb.server = ucp_ep_sockaddr_server_connect_cb;
+    uct_ep_params.disconnect_cb              = ucp_ep_sockaddr_disconnect_cb;
+
+    ucs_assert(ucp_worker_num_cm_cmpts(worker) == 1);
```

Comment:
pls use ucs_fatal with err message that ... is unsupported
also in 515

Reply:
replaced with ucs_assertv_always to encapsulate check

Reply:
ok